lib/index.d.ts

```diff
@@ -0,0 +1,465 @@
+// Type definitions for @hapi/jwt 2.0
+// Project: https://github.com/hapijs/jwt
+// Definitions by: Sergio Sánchez <https://github.com/ssanchezmarc>, Danilo Alonso <https://github.com/damusix>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 4.2
+
+import { Plugin, ResponseToolkit, Request, ResponseObject, ReqRef, ReqRefDefaults, MergeRefs } from '@hapi/hapi';
+
+declare module '@hapi/hapi' {
+    interface ServerAuth {
+        /**
+         * Function to define the server authentication strategy to be used.
+         *
+         * @param name string name to define the strategy.
+         * @param scheme jwt for this plugin.
+         * @param options jwt plugin options.
+         */
+        strategy(name: string, scheme: 'jwt', options?: HapiJwt.Options): void;
+    }
+}
+
+export declare namespace HapiJwt {
+    // Common definitions
+
+    type SupportedAlgorithm =
+        | 'RS256'
+        | 'RS384'
+        | 'RS512'
+        | 'PS256'
+        | 'PS384'
+        | 'PS512'
+        | 'ES256'
+        | 'ES384'
+        | 'ES512'
+        | 'HS256'
+        | 'HS384'
+        | 'HS512';
```

Comment:
As already mentioned:
```suggestion
        | 'HS512'
        | 'EdDSA';
```